featuretools/entityset/serialization.py

```diff
@@ -1,175 +1,228 @@
-        df.columns = [unicode(c) for c in df.columns]
-    return df, to_join
-
-
-def _write_pickle_entity_data(root, entity, metadata):
-    rel_filename = os.path.join(entity.id, 'data.p')
-    filename = os.path.join(root, rel_filename)
-    os.makedirs(os.path.join(root, entity.id))
-    pd_to_pickle(entity.data, filename)
-    metadata['entity_dict'][entity.id]['data_files'] = {
-        'data_filename': rel_filename,
-        'filetype': 'pickle',
-        'size': os.stat(filename).st_size
-    }
-    return metadata
+    path = os.path.abspath(path)
+    params = params or {}
+    if os.path.exists(path):
+        shutil.rmtree(path)
+    for dirname in [path, os.path.join(path, 'data')]:
+        os.makedirs(dirname)
+    description = entityset.create_data_description()
+    for entity in entityset.entities:
+        loading_info = write_entity_data(entity, path, **kwargs)
+        description['entities'][entity.id]['loading_info'].update(loading_info)
+        description['entities'][entity.id]['loading_info']['params'].update(params)
+    file = os.path.join(path, 'data_description.json')
+    with open(file, 'w') as file:
+        json.dump(description, file)
+
+
+def from_variable_description(description):
+    '''Deserialize variable from variable description.
+
+    Args:
+        description (dict) : Description of :class:`.Variable`.
+
+    Returns:
+        variable (Variable) : Returns :class:`.Variable`.
+    '''
+    is_string = isinstance(description['type'], str)
+    type = description['type'] if is_string else description['type']['value']
+    return VARIABLE_TYPES.get(type, VARIABLE_TYPES.get(None))
+
+
+def get_variable_types(description):
+    '''Returns variable types from entity description.
+
+    Args:
+        description (dict) : Description of :class:`.Entity`.
 
+    Returns:
+        variable_types (dict) : Returns dictionary where variable id maps to variable type.
+    '''
+    return {variable['id']: from_variable_description(variable) for variable in description['variables']}
+
+
+def from_entity_description(entityset, description, path=None):
+    '''Deserialize entity from entity description and add to entityset.
+
+    Args:
+        entityset (EntitySet) : Instance of :class:`.EntitySet` to add :class:`.Entity`.
+        description (dict) : Description of :class:`.Entity`.
+        path (str) : Root directory to serialized entityset.
+    '''
+    dataframe = read_entity_data(description, path=path)
+    variable_types = get_variable_types(description)
+    dtypes = description['loading_info']['properties']['dtypes']
+    entityset.entity_from_dataframe(
+        description['id'],
+        dataframe.astype(dtypes),
+        index=description.get('index'),
+        time_index=description.get('time_index'),
+        secondary_time_index=description['properties'].get('secondary_time_index'),
+        variable_types=variable_types)
+
+
+def from_relationship_description(entityset, description):
```

Comment:
this method should return the Relationship class

Reply:
Change applied in [this commit.](https://github.com/Featuretools/featuretools/pull/361/commits/2b1b4fb9d08feedc5a5f8ea1a9cd6516d4392749)